### ISSUE TITLE
Accessibility: Cannot use `x` buttons in `SearchableMultiDropdownField` via keyboard

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#2004]
The `x` buttons used to remove specific items and to clear all selections from a `SearchableMultiDropdownField` cannot be used via keyboard interaction.
<img width="1854" height="748" alt="Image" src="https://github.com/user-attachments/assets/9d63b0fe-39bb-4f1a-ab45-f3b90d4ee0d9" />

It's possible (using the left and right arrow buttons) to focus on the individual `x` buttons on each selected item, however these are `div` instead of `button` and pressing spacebar or enter doesn't activate the click handlers.

It doesn't appear to be possible to focus on the clear all `x` button (which is also a `div`).

This likely affects other dropdown fields as well.

**Comment by emteknetnz** [comment]
By default in react-select you can delete individual options with the backspace key. This should ideally be changed to space (activate).

Enter will currently submit the entire form the field is on which does seem like the correct behaviour and that's what will happen if you're focused on my fields e.g TextField

You cannot focus the 'delete all' button on the right using keyboard alone which is annoying, though this is possibly OK though since there is a work around available which is to just delete all the options one at a time. It seems unlikely that you'd have more than half a dozen options in one of these fields.

**Comment by emteknetnz** [comment]
I've spent a couple of hours trying to get spacebar to work, it really does not want to

I've tried:
- Handling the key press via `<Select onKeyDown={myHandleDownFunc}>` - however the event.target on keydown is not the 'x' button that was just pressed. Seems like this handler is simply fired too late to be useful for what we're doing.
- Passing in [custom components](https://react-select.com/components)
  - `MultiValueRemove` - however the `innerProps` only give you non-keyboard props such as onClick and onMouseDown, meaning we can't actually use this custom component to capture the key press
  - `Input` - using a custom Input just broke the component
  - `SelectContainer` - didn't work either
